plasmapy/plasma/grids.py

```diff
@@ -1105,16 +1129,24 @@ def volume_averaged_interpolator(
 
         # Get the indices that are equal to nan (fill values), then set
         # their values to 0. They will be over-written after the interpolation
-
-        # Nan array is shape [n] and is 1 if none of the indices for a
-        # position are NaN, and 0 otherwise.
-        nan_mask = np.where(np.isnan(np.sum(i, axis=1)), 0, 1)
+        nan_mask = np.isnan(i).any(axis=1)
         # Replace all NaNs temporarily with 0
         i = np.where(np.isnan(i), 0, i)
```

Comment:
```suggestion
        i[np.isnan(i)] = 0
```

Speed boost since you're setting element values instead of rebuilding and redefining an array.